editor/post-permalink/index.js

```diff
@@ -42,21 +46,68 @@ class PostPermalink extends Component {
 		}, 4000 );
 	}
 
+	onEditPermalink( event ) {
+		event.preventDefault();
+		this.setState( { editingSlug: true } );
+	}
+
+	onCancelEditPermalink( event ) {
+		event.preventDefault();
+		this.setState( { editingSlug: false } );
+	}
+
+	onSavePermalink( event ) {
+		event.preventDefault();
+		this.setState( { editingSlug: false } );
+	}
+
 	render() {
-		const { isNew, link } = this.props;
+		const { isNew, link, samplePermalink } = this.props;
+		const { showCopyConfirmation, editingSlug } = this.state;
 		if ( isNew || ! link ) {
 			return null;
 		}
 
+		let permalink = link,
+			viewLink = link;
+		if ( samplePermalink ) {
+			permalink = samplePermalink[ 0 ].replace( '%postname%', samplePermalink[ 1 ] );
+			viewLink += '&preview=true';
+		}
+
+		const prefix = permalink.replace( /[^/]+\/?$/, '' ),
+			slug = permalink.replace( /.*\/([^/]+)\/?$/, '$1' );
+
 		return (
 			<div className="editor-post-permalink">
 				<Dashicon icon="admin-links" />
 				<span className="editor-post-permalink__label">{ __( 'Permalink:' ) }</span>
-				<Button className="editor-post-permalink__link" href={ link } target="_blank">
-					{ link }
-				</Button>
-				<ClipboardButton className="button" text={ link } onCopy={ this.onCopy }>
-					{ this.state.showCopyConfirmation ? __( 'Copied!' ) : __( 'Copy' ) }
+				<span className="editor-post-permalink__link">
+					<span className="editor-post-permalink__prefix">
+						{ prefix }
+					</span>
+					{ ! editingSlug &&
+						<Button
+							className="editor-post-permalink__slug"
+							onClick={ this.onEditPermalink }
```

Comment:
We shouldn't offer the option to edit permalink if `%postname%` doesn't exist in the permalink structure.